zbuf/zval.go

```diff
@@ -73,28 +72,33 @@ func ZvalToZeekString(typ zng.Type, zv zcode.Bytes, isContainer bool, utf8 bool)
 			if err != nil {
 				return "error in ZvalToZeekString"
 			}
-			val, err := inner.New(v)
+			fld := ZvalToZeekString(inner, v, utf8)
 			if err != nil {
 				return "error in ZvalToZeekString"
 			}
-			fld := escape(val.String(), utf8)
 			// Escape the set separator after ZeekEscape.
 			_, _ = b.WriteString(strings.ReplaceAll(fld, ",", "\\x2c"))
+
 			if it.Done() {
 				break
 			}
 			_ = b.WriteByte(',')
 		}
-		s = b.String()
+		return b.String()
+	case *zng.TypeOfString:
+		if len(zv) == 1 && zv[0] == '-' {
```

Comment:
Nit: Maybe `string(zq) == "-"` if you find it clearer.

Reply:
I do, but thought that might allocate.

Reply:
It does not.